plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2201,24 +2211,36 @@ read_only::get_table_rows_result read_only::get_kv_table_rows_context( const rea
    // iterate the range
    auto walk_table_row_range = [&]( auto &itr, auto &end_itr, bool reverse ) {
       uint32_t actual_size = 0;
-      uint32_t key_size;
-      uint32_t val_size;
+
       const vector<char> &end_key = (reverse ? lb_key : ub_key);
       if( reverse ) {
          key_size = ub_key_size;
-         val_size = ub_value_size;
+         value_size = ub_value_size;
       } else {
          key_size = lb_key_size;
-         val_size = lb_value_size;
+         value_size = lb_value_size;
       }
       auto cmp = itr->kv_it_key_compare(end_key.data(), end_key.size());
       if( reverse ) cmp *= -1;
 
       unsigned int count = 0;
       for( count = 0; cur_time <= end_time && count < p.limit && cmp < 0; cur_time = fc::time_point::now() ) {
          row_value.clear();
-         row_value.resize(val_size);
-         status = itr->kv_it_value(offset, row_value.data(), val_size, actual_size);
+         row_value.resize(value_size);
+         status = itr->kv_it_value(offset, row_value.data(), value_size, actual_size);
+         EOS_ASSERT(status_lb == chain::kv_it_stat::iterator_ok, chain::contract_table_query_exception,  "Invalid key in ${t} ${i}", ("t", p.table)("i", p.index_name));
+
+         if (!is_primary_idx) {
+            std::swap(row_key, row_value);
+            auto success = kv_context.kv_get(p.code.to_uint64_t(), row_key.data(), row_key.size(), value_size);
+            if (success) {
+               row_value.resize(value_size);
+               actual_size = kv_context.kv_get_data(offset, row_value.data(), value_size);
+               EOS_ASSERT(value_size == actual_size, chain::contract_table_query_exception, "range query value size mismatch: ${s1} ${s2}", ("s1", value_size)("s2", actual_size));
+            } else {
+               EOS_ASSERT(value_size == actual_size, chain::contract_table_query_exception, "range query value size mismatch: ${s1} ${s2}", ("s1", value_size)("s2", actual_size));
```

Comment:
actual_size has not been initialized here. Do we need thus assert?

Reply:
changed assert